Resources Platform and Tools Tabs #4r7npy

mahamzayed opened the pull request:
# Description

The purpose of this PR is to wire up the platform and tools tabs on the resources page. Please note the following:

- I'm aware that you navigate to `http://localhost:3000/resources?type=platform` or `http://localhost:3000/resources?type=tools` independently on their own, no data will load. This is because the data for resources initially comes back on the resources tab and is available from there if you navigate through the three tabs on the page without refreshing. There is also no content type for `platforms` and `tools` in contentful, just `sparcAwards`. I think I have a way to fix this and can address if this is a bug that can't be overlooked.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Navigate to the Resources page and verify that the `platform` and `tools` tabs load data as expected (you can verify the resource type for each tab by selecting the `Resources` component and inspecting the data in Vue Dev Tools.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

cameronbaney commented:
@mahamzayed Is your plan to fix the bug about direct loading in a separate PR?

mahamzayed commented:
> @mahamzayed Is your plan to fix the bug about direct loading in a separate PR?

Yes I can do that just so we can get this merged. I'll create a ticket.

cameronbaney commented on pages/resources/index.vue:
```diff
@@ -55,21 +55,18 @@ const client = createClient()
-const ToolSearchResults = () =>
-  import('@/components/Resources/ToolSearchResults.vue')
 
 const resourcesResultsComponents = {
```
If all tabs are using the same component, there's no need for a dynamic component. Removing this logic will increase performance and redundant code.

mahamzayed commented:
@cameronbaney Addressed feedback. Didn't realize that I could pass in the resource type into the async call to get the data I needed. Thanks for pointing that out. 👍

cameronbaney commented on pages/resources/index.vue:
```diff
@@ -204,26 +182,31 @@ export default {
+      client
+        .getEntries(entries)
         .then(response => {
+          console.log('this is my response ', response)
```
```suggestion
```

cameronbaney commented on pages/resources/index.vue:
```diff
@@ -317,7 +296,7 @@ export default {
     overflow: auto;
     padding: 0;
-    margin-top: 3.625rem;
+    // margin-top: 3.625rem;
```
```suggestion
```

cameronbaney commented on pages/resources/index.vue:
```diff
@@ -260,9 +238,10 @@ export default {
     padding-bottom: 0;
     p {
-      font-size: 20px;
+      font-size: 17px;
```
```suggestion
      font-size: 1.0625em;
```

cameronbaney commented on pages/resources/index.vue:
```diff
@@ -128,7 +119,9 @@ export default {
      */
     currentResourceCount: function() {
-      return this.resourceData.skip !== 0
+      return this.resourceData.limit > this.resourceData.total
```
This looks like a bad merge since it's returning two statements.

mahamzayed replied:
No the logic here is correct. The platform tab for example only has 11 resources instead of 12. This checks if the limit of the data returned is larger than that of the total result set, and if it is than it should return the total number instead of the limit. So instead of it saying `Showing 12 of 11 resources` it says `showing `11 of 11 resources`. You can chain multiple `else if` blocks in tierary operators which is why it looks weird here.

cameronbaney replied:
That's interesting, I didn't know that.

mahamzayed replied:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator 

Yeah just learned it the other day! Pretty cool 👍

cameronbaney commented on pages/resources/index.vue:
```diff
@@ -197,15 +182,30 @@ export default {
+        this.$route.query.type === 'Platform' ||
+        this.$route.query.type === 'Tool'
+      ) {
+        entries = {
```
nit: Another way to do this would be to `Object.assign()` the `entries` object with the new keys.

mahamzayed commented:
> The table heading at the top says "Showing 5 of **5 resources**" for all types. Should the word "resources" be based off what what type you're showing? If not, looks good.

@cameronbaney Good question. I'll ask Annelise to see what her thoughts are.

mahamzayed commented:
@cameronbaney Addressed feedback
